### PR TITLE
Technical features

### PR DESCRIPTION
**Pull request by augustocollado**
### Description

This PR adds the technical features article.

### Checklist

- [ ] If this page requires a disclaimer, I have added one
- [ ] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: learn/tanssi/technical.md [hunk 1]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
+In Tanssi terms, Appchains are called ContainerChains, similar to the concept coined in [Docker](https://www.docker.com){target=blank}, allowing teams to focus on the product while alleviating deployment-related issues.
 
-- **Weak & Inefficient Security** - Appchains commonly suffer from having small or weak validator sets. Early-stage Appchains don't have enough economic guarantees to power a robust consensus mechanism. Moreover, paying for full blockchain capacity is inefficient when only a fraction of it is needed
+In this article, we discuss the following technical aspects of how Tanssi works:
 
-- **Cross-Chain and Interoperability** - Appchains inherently don't have cross-chain capabilities to connect to other blockchain ecosystems. Furthermore, developing interoperability solutions requires specialized expertise and meticulous implementation
+- **Collators Assignment**
+- **Block Production**
+- **Implementing Tanssi Protocol in your ContainerChain**
+- **Deploy a new ContainerChain**
```
Comment (eshaben) [inline learn/tanssi/technical.md:21]:
```suggestion
- **Collator assignment**
- **Block production**
- **Implementing the Tanssi protocol in your ContainerChain**
- **Deploying a new ContainerChain**
```

Comment (eshaben) [reply learn/tanssi/technical.md:21]: I feel like the last two aren't really "how Tanssi works" but what is needed to integrate Tanssi - so maybe rephrase the descriptive line above this list to just be the technical aspects related to Tanssi

File: learn/tanssi/technical.md [hunk 2]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
 
-- **Slow Time to Market** - Appchain's infrastructure complexities diver developer focus from application logic, which is the key driver for intuitive interfaces and seamless user experience, which is critical for adoption
+## Collators Assignment {: #collators-assignment } 
 
-## What Tanssi Provides
+Tanssi Protocol manages a set of collators and assigns them to provide block production services to the active ContainerChains and the Tanssi Network itself.
```
Comment (eshaben) [inline learn/tanssi/technical.md:25]: Protocol and Network shouldn't be capitalized
```suggestion
The Tanssi protocol manages a set of collators and assigns them to provide block production services to the active ContainerChains and the Tanssi network itself.
```

File: learn/tanssi/technical.md [hunk 3]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
 
-Tanssi addresses the most common Appchain problems by:
+The assignment algorithm will start distributing the available collators, serving first the Tanssi Network and then the ContainerChains, ordered by the registration date, being served on a first come first served basis. Once the assignment is made, it will be upheld for at least one session.
```
Comment (eshaben) [inline learn/tanssi/technical.md:27]:
```suggestion
The assignment algorithm will start distributing the available collators, serving first the Tanssi network and then the ContainerChains, ordered by the registration date, on a first-come, first-served basis. Once the assignment is made, it will be upheld for at least one session.
```

What do you mean by a session?

File: learn/tanssi/technical.md [hunk 4]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
-3. Governance and Upgrades: Polkadot introduces an on-chain governance model, empowering token holders to participate in decision-making and allowing for efficient upgrades and protocol improvements.
-4. Security: In smaller blockchain networks, security can be a concern due to lower levels of mining or staking participation. Polkadot’s model of shared security guarantees the same high security level even for the smaller parachain within the ecosystem.
-5. Customizability/flexibility: Polkadot's architecture encourages the creation of sovereign blockchains with high specialization, enabling developers to design chains with specific features and functionalities optimized for the unique requeriments of the use case.
+We can consider Tanssi Network and the ContainerChains as sibling chains, meaning that there is no hierarchical dependency whatsoever, and we will discuss their responsibility and how they interact with each other through the relay chain in the following sections.
```
Comment (eshaben) [inline learn/tanssi/technical.md:44]:
```suggestion
The Tanssi network and the ContainerChains can be considered as sibling chains, meaning that there is no hierarchical dependency whatsoever. Their responsibility and how they interact with each other through the relay chain will be covered in the following sections.
```

Try to avoid using "we" or "let's" unless it's an informal writing piece like a tutorial

File: learn/tanssi/technical.md (hunk 2 above)
Comment (eshaben) [reply learn/tanssi/technical.md:25]: If you're saying "Tanssi protocol" or "Tanssi network" you should use "the" before, so "the Tanssi protocol", etc. Or you can just say "Tanssi" and don't need "the".

Lots of places throughout this page don't use "the" before calling it a protocol or network, please fix these instances too

File: learn/tanssi/technical.md [hunk 5]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
\ No newline at end of file
+### Tanssi Network {: #tanssi-newtwork } 
+
+As previously discussed, Tanssi Protocol assigns collators to Tanssi Network itself and the ContainerChains, and the result of this assignment is stored within the chain state.
+
+Another important piece of information that Tanssi stores is the latest header for every ContainerChain. This data is read from the relay chain and, being stored in every Tanssi block, it allows the protocol to keep track of the state in every chain and also to identify and reward accordingly the collator that produced their last block.
+
+### ContainerChain {: #containerchain } 
+
+As a collator node assigned to a ContainerChain deployed in Tanssi, and having a built-in Tanssi node functionality, it is technically feasible to read the state from the relay chain and the state from the Tanssi Network.
```
Comment (eshaben) [inline learn/tanssi/technical.md:63]:
```suggestion
As a collator node assigned to a ContainerChain deployed in Tanssi has built-in Tanssi node functionality, it is technically feasible to read the state from the relay chain and the state from the Tanssi Network.
```

File: learn/tanssi/technical.md [hunk 6]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
+
+Leveraging this ability to access the states, the current collator with the authority to produce a block will read the latest block produced in the relay chain which contains the state root of the latest block produced in Tanssi. With this state root, it will proceed to read the state in Tanssi, and include in the block of the ContainerChain the latest state root of Tanssi Network, the current set of collators assigned to the ContainerChain, and its public signature, allowing Tanssi to know who produced the block and reward the collator.
```
Comment (eshaben) [inline learn/tanssi/technical.md:65]:
```suggestion
Leveraging this ability to access the states, the current collator with the authority to produce a block will read the latest block produced in the relay chain, which contains the state root of the latest block produced in Tanssi. With this state root, it will proceed to read the state in Tanssi, and include in the block of the ContainerChain the latest state root of Tanssi Network, the current set of collators assigned to the ContainerChain, and its public signature, allowing Tanssi to know who produced the block and reward the collator.
```

File: learn/tanssi/technical.md [hunk 7]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
+In this article, we discuss the following technical aspects of how Tanssi works:
 
-- **Cross-Chain and Interoperability** - Appchains inherently don't have cross-chain capabilities to connect to other blockchain ecosystems. Furthermore, developing interoperability solutions requires specialized expertise and meticulous implementation
+- **Collators Assignment**
+- **Block Production**
+- **Implementing Tanssi Protocol in your ContainerChain**
+- **Deploy a new ContainerChain**
 
-- **Slow Time to Market** - Appchain's infrastructure complexities diver developer focus from application logic, which is the key driver for intuitive interfaces and seamless user experience, which is critical for adoption
+## Collators Assignment {: #collators-assignment }
```
Comment (eshaben) [inline learn/tanssi/technical.md:23]:
```suggestion
## Collator Assignment {: #collators-assignment } 
```

File: learn/tanssi/technical.md [hunk 8]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
+
+### ContainerChain {: #containerchain } 
+
+As a collator node assigned to a ContainerChain deployed in Tanssi, and having a built-in Tanssi node functionality, it is technically feasible to read the state from the relay chain and the state from the Tanssi Network.
+
+Leveraging this ability to access the states, the current collator with the authority to produce a block will read the latest block produced in the relay chain which contains the state root of the latest block produced in Tanssi. With this state root, it will proceed to read the state in Tanssi, and include in the block of the ContainerChain the latest state root of Tanssi Network, the current set of collators assigned to the ContainerChain, and its public signature, allowing Tanssi to know who produced the block and reward the collator.
+
+Once the block is completed with the ContainerChain transactions, it will be proposed, validated, and finalized by the relay chain, which will include the block's header in the relay chain block.
+
+## Implementing Tanssi Protocol in your ContainerChain
```
Comment (eshaben) [inline learn/tanssi/technical.md:69]:
```suggestion
## Implementing the Tanssi Protocol in your ContainerChain {: #implementing-tanssi-in-your-containerchain }
```

File: learn/tanssi/technical.md [hunk 9]
```diff
@@ -3,51 +3,83 @@ title: Technical Features
...
+
+-**Author Inherent pallet** - this pallet is necessary to allow the collator to include in the block its identity and be recognized and awarded as the block producer
+
+It is important to note that both pallets include the mentioned data in the block using Inherents, which are a special form of transaction that only the block producer can include.
+
+## Deploy a ContainerChain {: #deploy-a-containerchain } 
+
+After building on top of one of the provided [Appchain Templates](/learn/tanssi/templates) and finishing the development process, developers are ready to deploy their ContainerChain in Tanssi.
+
+This a fairly straightforward step, where the teams need only to generate and upload the [chain specification](https://docs.substrate.io/build/chain-spec/){target=blank} to the Tanssi orchestrator.
```
Comment (eshaben) [inline learn/tanssi/technical.md:83]: Is the only purpose of the Tanssi orchestrator to assign a set of collators? Since this is the first time mentioning it, it might be nice to add a sentence or two to briefly describe what the orchestrator is

Comment (eshaben) [inline learn/tanssi/technical.md:83]:
```suggestion
This is a fairly straightforward step, where the teams only need to generate and upload the [chain specification](https://docs.substrate.io/build/chain-spec/){target=blank} to the Tanssi orchestrator.
```

**Review by eshaben — CHANGES_REQUESTED** [review]
Nice job! Diagrams look great. Mostly little things! 🙂

File: learn/tanssi/technical.md (hunk 9 above)
Comment (augustocollado) [reply learn/tanssi/technical.md]: I started using "tanssi orchestrator" as a synonym for the tanssi network. I changed it back :)

**Review by eshaben — APPROVED** [review]
Looks good! I mostly looked at structural/formatting/grammar-related items. If Alberto doesn't review this, I'll probably take another look to make sure content is all good

File: learn/tanssi/technical.md [hunk 10]
```diff
@@ -1,53 +1,91 @@
...
 
-Polkadot -and the parachains- are built with an open-source blockchain development framework called Substrate, which is bases on Rust programming language.
+Among many other responsibilities, the relay chain validates and finalizes the blocks produced by any chain participating in the ecosystem (including the ContainerChains and the Tanssi network), storing the most recent headers for each block of each chain.
 
-Substrate makes it easy for teams to develop blockchains, providing pre built core functinalities such as consensus mechanisms, staking & governance, account and assets management, networking and many more ready to use modules and libraries.
+The Tanssi protocol leverages this feature, relying on the relay chain as a means to provide the necessary data to both, Tanssi and its ContainerChains, allowing them to collaborate and validate the correctness of the block production service.
 
-The core principles of Substrate are ease of use and performance, allowing teams to create specialized blockchains for any use case, with high level of performance, flexibility, and robustness.
\ No newline at end of file
+![Relay chain](/images/learn/tanssi/technical/technical-3.png)
```
Comment (eshaben) [inline learn/tanssi/technical.md:55]: Actually, there are some things that need to be centered in this diagram. The text doesn't seem to be aligned in the middle in a few different places i.e., "Tannsi ContainerChain" doesn't look centered in the orange square, "Root State 0x265d1" looks like it is more to the left than centered

File: learn/tanssi/technical.md [hunk 11]
```diff
@@ -1,53 +1,91 @@
...
+
+### Tanssi Network {: #tanssi-newtwork } 
+
+As previously discussed, the Tanssi protocol assigns collators to the Tanssi network itself and the ContainerChains, and the result of this assignment is stored within the chain state.
+
+Another important piece of information that Tanssi stores is the latest header for every ContainerChain. This data is read from the relay chain and, being stored in every Tanssi block, it allows the protocol to keep track of the state in every chain and also to identify and reward accordingly the collator that produced their last block.
+
+![Tanssi Network](/images/learn/tanssi/technical/technical-4.png)
```
Comment (eshaben) [inline learn/tanssi/technical.md:63]: I think instead of the Tanssi assigned collators and listed C1, C2, C3 like text, it would be cool if they were in the same format as they are in the first image with the yellow circle background

"ContainerChainStateRoot" doesn't look like it's centered in the teal border

You haven't used "-" except here, maybe just remove the dashes here in front of "Assigns collators" and "Stores ContainerChain state..."

File: learn/tanssi/technical.md [hunk 12]
```diff
@@ -1,53 +1,91 @@
...
+
+### ContainerChain {: #containerchain } 
+
+As a collator node assigned to a ContainerChain deployed in Tanssi has built-in Tanssi node functionality, it is technically feasible to read the state from the Tanssi network and the blocks from the relay chain.
+
+Leveraging this ability to access the states, the current collator with the authority to produce a block will read the latest block produced in the relay chain, which contains the state root of the latest block produced in Tanssi. With this state root, it will proceed to read the state in Tanssi, and include in the block of the ContainerChain the latest state root of the Tanssi network, the current set of collators assigned to the ContainerChain, and its public signature, allowing Tanssi to know who produced the block and reward the collator.
+
+Once the block is completed with the ContainerChain transactions, it will be proposed, validated, and finalized by the relay chain, which will include the block's header in the relay chain block.
+
+![ContainerChain collation](/images/learn/tanssi/technical/technical-5.png)
```
Comment (eshaben) [inline learn/tanssi/technical.md:73]: Same comment here as above with the assigned collators, it would be cool if it had the yellow circle background.

The Tanssi State Root State 0x265d1 - is that meant to say "State" twice? Also it looks like that needs to be a little wider. The border around "ContainerChain Assigned Collators" is cutting into some of the letters ever so slightly and it would just look better with a little extra white space

Comment (eshaben) [reply learn/tanssi/technical.md:73]: Inconsistent capitalization with "Relay Chain Block" and "ContainerChain block" block should probably be capitalized in both cases

**Review by eshaben — CHANGES_REQUESTED** [review]
Sorry I don't know how I missed the diagrams, but left some comments about those!

File: learn/tanssi/technical.md (hunk 10 above)
Comment (eshaben) [reply learn/tanssi/technical.md:55]: I know customizing these diagrams for dark and light backgrounds is probably a hassle, but i'm not a huge fan of the dark version of this image due to the white background in the Relay Chain Block - can you possibly just remove the white background and make all of the text there pink?

Comment (eshaben) [reply learn/tanssi/technical.md:55]: Also professional nitpicker sorry... in the Relay Chain Block box ContainerChain state root and Tanssi state root should be swapped because on the left side Tanssi is on the top

I know I just left a comment about changing the text to pink, but as I keep looking at it, I actually think like you should use the same rectangle and border on the left side, for example the teal background and text copied over to the right side and replace "Tanssi state root" maybe just alter the text so it says "Tanssi State Root" instead of "Block Root State". Same with the ContainerChain.

Also it should be "State Root" instead of "Root State" and this comment applies to all the diagrams

File: learn/tanssi/technical.md (hunk 12 above)
Comment (augustocollado) [reply learn/tanssi/technical.md]: State was twice on purpose, but yes, It looks a bit confusing. For the sake of clarity, I've just removed the word!

File: learn/tanssi/technical.md [hunk 13]
```diff
@@ -1,53 +1,93 @@
...
 
-Polkadot -and the parachains- are built with an open-source blockchain development framework called Substrate, which is bases on Rust programming language.
+Among many other responsibilities, the relay chain validates and finalizes the blocks produced by any chain participating in the ecosystem (including the ContainerChains and the Tanssi network), storing the most recent headers for each block of each chain.
 
-Substrate makes it easy for teams to develop blockchains, providing pre built core functinalities such as consensus mechanisms, staking & governance, account and assets management, networking and many more ready to use modules and libraries.
+The Tanssi protocol leverages this feature, relying on the relay chain as a means to provide the necessary data to both, Tanssi and its ContainerChains, allowing them to collaborate and validate the correctness of the block production service.
 
-The core principles of Substrate are ease of use and performance, allowing teams to create specialized blockchains for any use case, with high level of performance, flexibility, and robustness.
\ No newline at end of file
+![Relay chain](/images/learn/tanssi/technical/technical-3.png)
```
Comment (eshaben) [inline learn/tanssi/technical.md:55]: The font is super small for "Tanssi state root" on the right side, can you please make it a little bit bigger? The colors look good!!

File: learn/tanssi/technical.md [hunk 14]
```diff
@@ -1,53 +1,93 @@
...
+
+### Tanssi Network {: #tanssi-newtwork } 
+
+As previously discussed, the Tanssi protocol assigns collators to the Tanssi network itself and the ContainerChains, and the result of this assignment is stored within the chain state.
+
+Another important piece of information that Tanssi stores is the latest header for every ContainerChain. This data is read from the relay chain and, being stored in every Tanssi block, it allows the protocol to keep track of the state in every chain and also to identify and reward accordingly the collator that produced their last block.
+
+![Tanssi Network](/images/learn/tanssi/technical/light-technical-4.png#only-light)
+![Tanssi Network](/images/learn/tanssi/technical/dark-technical-4.png#only-dark)
```
Comment (eshaben) [inline learn/tanssi/technical.md:64]: The teal border around the ContainerChain state root is a bit more rounded than those in the 5th image.

And the items on the right side under Tanssi state don't have a `:` at the end like they do in the other images

Comment (eshaben) [reply learn/tanssi/technical.md:64]: If you wanted to have only one image and not a light and dark, you could change the font colors so they stay the same. I think the pink and teal look good on both dark and light backgrounds. But I'll leave that up to you!

File: learn/tanssi/technical.md [hunk 15]
```diff
@@ -1,53 +1,93 @@
...
+### ContainerChain {: #containerchain } 
+
+As a collator node assigned to a ContainerChain deployed in Tanssi has built-in Tanssi node functionality, it is technically feasible to read the state from the Tanssi network and the blocks from the relay chain.
+
+Leveraging this ability to access the states, the current collator with the authority to produce a block will read the latest block produced in the relay chain, which contains the state root of the latest block produced in Tanssi. With this state root, it will proceed to read the state in Tanssi, and include in the block of the ContainerChain the latest state root of the Tanssi network, the current set of collators assigned to the ContainerChain, and its public signature, allowing Tanssi to know who produced the block and reward the collator.
+
+Once the block is completed with the ContainerChain transactions, it will be proposed, validated, and finalized by the relay chain, which will include the block's header in the relay chain block.
+
+![ContainerChain collation](/images/learn/tanssi/technical/light-technical-5.png#only-light)
+![ContainerChain collation](/images/learn/tanssi/technical/dark-technical-5.png#only-dark)
```
Comment (eshaben) [inline learn/tanssi/technical.md:75]: Same comment as above. If you want to have only one image, you could use the same font colors pink/teal for both light and dark backgrounds, but again, up to you!

**Review by eshaben — CHANGES_REQUESTED** [review]
Just a couple tiny things! Looking good though!!

File: learn/tanssi/technical.md (hunk 15 above)
Comment (augustocollado) [reply learn/tanssi/technical.md]: Unified dark & light images :)

**Review by eshaben — APPROVED** [review]
LGTM! ✨

File: learn/tanssi/technical.md [hunk 16]
```diff
@@ -1,53 +1,91 @@
...
+![Sibling Chains](/images/learn/tanssi/technical/dark-technical-2.png#only-dark)
 
-## The technology behind
+### Relay Chain {: #relay-chain } 
 
-Polkadot -and the parachains- are built with an open-source blockchain development framework called Substrate, which is bases on Rust programming language.
+Among many other responsibilities, the relay chain validates and finalizes the blocks produced by any chain participating in the ecosystem (including the ContainerChains and the Tanssi network), storing the most recent headers for each block of each chain.
 
-Substrate makes it easy for teams to develop blockchains, providing pre built core functinalities such as consensus mechanisms, staking & governance, account and assets management, networking and many more ready to use modules and libraries.
+The Tanssi protocol leverages this feature, relying on the relay chain as a means to provide the necessary data to both, Tanssi and its ContainerChains, allowing them to collaborate and validate the correctness of the block production service.
```
Comment (albertov19) [inline learn/tanssi/technical.md:53]: Be more specific here, remember that this is the technical section that is not meant to be high level.

For example, state that this means that all chains have a shared source of truth, and that sibling chains have access to the latest state roots of other chains as they ingest this from the relay chain (and not sibling chains directly!)

File: learn/tanssi/technical.md [hunk 17]
```diff
@@ -1,53 +1,91 @@
...
+
+![Tanssi Network](/images/learn/tanssi/technical/technical-4.png)
+
+### ContainerChain {: #containerchain } 
+
+As a collator node assigned to a ContainerChain deployed in Tanssi has built-in Tanssi node functionality, it is technically feasible to read the state from the Tanssi network and the blocks from the relay chain.
+
+Leveraging this ability to access the states, the current collator with the authority to produce a block will read the latest block produced in the relay chain, which contains the state root of the latest block produced in Tanssi. With this state root, it will proceed to read the state in Tanssi, and include in the block of the ContainerChain the latest state root of the Tanssi network, the current set of collators assigned to the ContainerChain, and its public signature, allowing Tanssi to know who produced the block and reward the collator.
```
Comment (albertov19) [inline learn/tanssi/technical.md:69]: This is very clear! Nice!

I think it is important to point out that even tho the node can read state from Tanssi, the relay chain will always contain the proof that no one is cheating right?

So even if you modify the client to say that you are always the chosen block producer, that will not match with the state proof submitted by tanssi in this case

File: learn/tanssi/technical.md [hunk 18]
```diff
@@ -1,53 +1,91 @@
...
+
+Once the block is completed with the ContainerChain transactions, it will be proposed, validated, and finalized by the relay chain, which will include the block's header in the relay chain block.
+
+![ContainerChain collation](/images/learn/tanssi/technical/technical-5.png)
+
+## Implementing the Tanssi Protocol in your ContainerChain {: #implementing-tanssi-in-your-containerchain }
+
+To make your Appchain Tanssi compliant and ready to become a deployed ContainerChain, adding references to the following two pallets is required:
```
Comment (albertov19) [inline learn/tanssi/technical.md:77]: Not pallets! Modules!!

**Review by albertov19 — CHANGES_REQUESTED** [review]
Great job! Left some feedback, let's discuss

File: learn/tanssi/technical.md [hunk 19]
```diff
@@ -1,53 +1,91 @@
...
+The following picture shows an example of how the algorithm works by distributing ten available collators, with a minimum threshold of three collators for the Tanssi network and two collators for each of the ContainerChains.
 
-- **Key Integrations** - Appchains built with Tanssi with have access to key infrastuctural components alongisde block production. Because these Appchains are built with [Substrate](https://substrate.io/){target=_blank}, crucial components like wallets, block explorers, indexers, RPC providers, and others are supported out of the box. Furthermore, Appchains can leverage multiple interoperability protocols to connect to other blockchain ecosystems like Ethereum, Avalance, Binance Smart Chain and more
+![Collators Assignment Algorithm](/images/learn/tanssi/technical/light-technical-1.png#only-light)
```
[the remark this answers is not in the document]
Comment (augustocollado) [reply learn/tanssi/technical.md:46]: Included more info in the section role of the relay chain explaining with more detail the interactions.
Also, included a diagram for the sessions

File: learn/tanssi/technical.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (augustocollado) [reply learn/tanssi/technical.md:59]: Simplified the diagram, wdyt ?

File: learn/tanssi/technical.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (augustocollado) [reply learn/tanssi/technical.md:67]: Also simplified

File: learn/tanssi/technical.md [hunk 20]
```diff
@@ -1,53 +1,95 @@
...
 
-- **Complex Infrastructure Management** - Appchain deployments typically demand handling numerous infrastructural components like block producers, validators, wallets, block explorers, indexers, RPC endpoints, and more. This is both time and resource consuming
+In Tanssi terms, Appchains are called ContainerChains, similar to the concept coined in [Docker](https://www.docker.com){target=_blank}, allowing teams to focus on the product while alleviating deployment-related issues. In this analogy, the Tanssi network resembles [Kubernetes](https://kubernetes.io){target=_blank}, in its role as orchestrator, managing resources to guarantee the liveness and performance of the ContainerChains.
 
-- **Weak & Inefficient Security** - Appchains commonly suffer from having small or weak validator sets. Early-stage Appchains don't have enough economic guarantees to power a robust consensus mechanism. Moreover, paying for full blockchain capacity is inefficient when only a fraction of it is needed
+In this article, we discuss how to build and deploy your own Modular Blockchain and the following technical aspects of the Tanssi protocol:
```
Comment (albertov19) [inline learn/tanssi/technical.md:16]: We are not stating how to build and deploy your own Modular Blockchain right? Just explaining the technical aspects of Tanssi in more depth

File: learn/tanssi/technical.md [hunk 21]
```diff
@@ -1,53 +1,95 @@
...
 
-- **Key Integrations** - Appchains built with Tanssi with have access to key infrastuctural components alongisde block production. Because these Appchains are built with [Substrate](https://substrate.io/){target=_blank}, crucial components like wallets, block explorers, indexers, RPC providers, and others are supported out of the box. Furthermore, Appchains can leverage multiple interoperability protocols to connect to other blockchain ecosystems like Ethereum, Avalance, Binance Smart Chain and more
+The Tanssi protocol manages a set of collators and assigns them to provide block production services to the active ContainerChains and the Tanssi network itself.
 
+The assignment algorithm will start distributing the available collators, serving first the Tanssi network and then the ContainerChains, ordered by the registration date, on a first-come, first-served basis. Once the assignment is made, it will be upheld for at least one session, which represents a period measured in blocks that has a constant set of collators. In the provided [templates](/learn/tanssi/templates), the default session duration is set to 1800 blocks, which translates to (roughly) six hours.
```
Comment (albertov19) [inline learn/tanssi/technical.md:36]:
```suggestion
The assignment algorithm will start distributing the available collators, serving first the Tanssi network and then the ContainerChains, ordered by the registration date, on a first-come, first-served basis. Once the assignment is made, it will be upheld for at least one session, which represents a period measured in blocks that has a constant set of collators. In the provided [templates](/learn/tanssi/templates), the default session duration is set to 1800 blocks, which  with an average block time of 12 seconds, it translates to (roughly) six hours.
```

File: learn/tanssi/technical.md [hunk 22]
```diff
@@ -1,53 +1,95 @@
...
-4. Security: In smaller blockchain networks, security can be a concern due to lower levels of mining or staking participation. Polkadot’s model of shared security guarantees the same high security level even for the smaller parachain within the ecosystem.
-5. Customizability/flexibility: Polkadot's architecture encourages the creation of sovereign blockchains with high specialization, enabling developers to design chains with specific features and functionalities optimized for the unique requeriments of the use case.
+![Sessions](/images/learn/tanssi/technical/technical-6.png)
 
-The Polkadot Relay Chain serves as the main network that coordinates the overall operation of the platform providing shared security, consensus, and interoperability amongst the connected parachains, to accelerate the development of the web3.
+The following picture shows an example of how the algorithm works by distributing ten available collators, with a minimum threshold of three collators for the Tanssi network and two collators for each of the ContainerChains.
```
Comment (albertov19) [inline learn/tanssi/technical.md:44]: If you think about it, the picture is not showing how the algorithm works, just the result of the algorithm working hehe. Maybe change the wording a bit?

File: learn/tanssi/technical.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (albertov19) [reply learn/tanssi/technical.md:59]: Better! My only comment is that it is not clear that then ContainerChains have access to Tanssi's latest state root has etc.

File: learn/tanssi/technical.md [hunk 23]
```diff
@@ -1,53 +1,95 @@
...
+
+![Relay chain](/images/learn/tanssi/technical/technical-3.png)
+
+### The Role of the Tanssi Network {: #tanssi-newtwork } 
+
+As previously discussed, the Tanssi protocol assigns collators to the Tanssi network itself and the ContainerChains, and the result of this assignment is stored within the chain state.
+
+Another important piece of information that Tanssi stores is the latest header for every ContainerChain. This data is read from the relay chain and, being stored in every Tanssi block, it allows the protocol to keep track of the state in every chain and also to identify and reward accordingly the collator that produced their last block.
+
+![Tanssi Network](/images/learn/tanssi/technical/technical-4.png)
```
Comment (albertov19) [inline learn/tanssi/technical.md:67]: Better! However, I thought that the point of tanssi reading the container chain state root hash from ther elay chain was to be able to verify that CollatorA (for example) produced the block. So even if you write about this, it would be great to add that in the image also (in the right hand side table)

File: learn/tanssi/technical.md [hunk 24]
```diff
@@ -1,53 +1,95 @@
...
+
+Leveraging this ability to access the states, the current collator with the authority to produce a block will read the latest block produced in the relay chain, which contains the state root of the latest block produced in Tanssi. With this state root, it will proceed to read the state in Tanssi, and include in the block of the ContainerChain the latest state root of the Tanssi network, the current set of collators assigned to the ContainerChain, and its public signature, allowing Tanssi to know who produced the block and reward the collator.
+
+Once the block is completed with the ContainerChain transactions, it will be proposed as a candidate and handed over to the relay chain validators, which will ensure that the included state proofs match the state proofs from the latest state of Tanssi (preventing unauthorized collation) and that the transactions produced valid state transitions. Having verified the work of the collator, the relay chain will finalize the proposed block, including its candidate receipt in the relay chain block.
+
+![ContainerChain collation](/images/learn/tanssi/technical/technical-5.png)
```
Comment (albertov19) [inline learn/tanssi/technical.md:77]: Stay consistent with the design of the image. This one you used a white background for the boxes

**Review by albertov19 — CHANGES_REQUESTED** [review]
Looking good! Some minor nitpicks :)

File: learn/tanssi/technical.md (hunk 23 above)
Comment (augustocollado) [reply learn/tanssi/technical.md]: Rearranged the relay chain image, closer to the text that describes it at the beginning, added a line about the collator running also a tanssi node, and the last block producer to the image

**Review by albertov19 — APPROVED** [review]
LGTM! Good job
